### PR TITLE
Fix unhandled exception in validateInputs in RSIQ

### LINKED ISSUE #22933: Unexpected exception in algorithm dialog with IndexProperty
### Expected behavior
When you run `ReflectometrySumInQ` on an invalid workspace type it should open the algorithm dialog and highlight the error.

### Actual behavior
Mantid throws an unexpected exception and the algorithm dialog does not open.

### Steps to reproduce the behavior
- Run this script:
```
LoadNexus(Filename='/archive/ndxpolref/Instrument/data/cycle_15_2/POLREF00014891.nxs', OutputWorkspace='TOF_14891')
ReflectometryReductionOneAuto(InputWorkspace='TOF_14891_1', ThetaIn=0.25037454461736225, MomentumTransferMin=0.0036648166871465389, MomentumTransferStep=0.041685322949264948, MomentumTransferMax=0.053981076801802731, ScaleFactor=1, OutputWorkspaceBinned='IvsQ_binned_14891_1', OutputWorkspace='IvsQ_14891_1', OutputWorkspaceWavelength='IvsLam_14891_1')
ReflectometryReductionOneAuto(InputWorkspace='TOF_14891_2', ThetaIn=0.25037454461736225, MomentumTransferMin=0.0036648166871465389, MomentumTransferStep=0.041685322949264948, MomentumTransferMax=0.053981076801802731, ScaleFactor=1, OutputWorkspaceBinned='IvsQ_binned_14891_2', OutputWorkspace='IvsQ_14891_2', OutputWorkspaceWavelength='IvsLam_14891_2')
GroupWorkspaces(InputWorkspaces='IvsLam_14891_1,IvsLam_14891_2', OutputWorkspace='IvsLam_14891')
GroupWorkspaces(InputWorkspaces='IvsQ_14891_1,IvsQ_14891_2', OutputWorkspace='IvsQ_14891')
GroupWorkspaces(InputWorkspaces='IvsQ_binned_14891_1,IvsQ_binned_14891_2', OutputWorkspace='IvsQ_binned_14891')
ReflectometrySumInQ(InputWorkspace='TOF_14891')
```
The the script window gives an error from `ReflectometrySumInQ`, as you'd expect:
```
ValueError: Invalid value for property InputWorkspace (MatrixWorkspace) "TOF_14891": The workspace must have units of Wavelength
```
- Now try opening the algorithm dialog for `ReflectometrySumInQ` on the same workspace. You get:
```
Unexpected exception: Invalid workspace type provided to IndexProperty. Must be convertible to MatrixWorkspace.
```

This seems to be to do with the  `IvsLam_` workspace group - if you delete this then it works.

### Platforms affected
Seen on Ubuntu 16.04. Not reproducible on Windows 7.

### PR DESCRIPTION
**Pull request by gemmaguest**
**Description of work.**

This PR adds a temporary fix for an unhandled exception in `ReflectometrySumInQ::validateInputs`. The exception occurs when opening AlgorithmDialog on a group workspace. This first calls validateInputs on the group workspace for the dialog validation, before calling it on the individual workspaces when executing the algorithm. Workspace groups are not currently handled. Adding full validation for groups is beyond this PR so this fix just skips validation to avoid the error.

Fixes #22933

**To test:**

- Follow the instructions in the linked issue and check you no longer get an unhandled exception.
- Check the algorithm runs ok for valid inputs for both group and non-group workspaces.
- Try entering some invalid values, e.g. set WorkspaceIndexSet=`4-8` and BeamCentre=`3`. For a non-group workspace this gives a validation error on the dialog. For a group workspace the validation is skipped so the algorithm runs and shows an error in the log.

Does this update require release notes?
- [ ] Yes
- [x] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by gemmaguest** [comment]
I wanted to check if the workspace type is a group but had trouble getting something sensible out of the property manager, so for now I've just caught the exception.

**Comment by NickDraper** [comment]
warning in ubuntu test is already fixed on the release-next branch

File: Framework/Algorithms/src/ReflectometrySumInQ.cpp [hunk 1]
```diff
@@ -269,8 +269,18 @@ std::map<std::string, std::string> ReflectometrySumInQ::validateInputs() {
...
-      getWorkspaceAndIndices<API::MatrixWorkspace>(Prop::INPUT_WS);
+
+  // validateInputs is called on the individual workspaces when the algorithm
+  // is executed, it but may get called on a group from AlgorithmDialog. This
+  // isn't handled in getWorkspaceAndIndices. We should fix this properly but
+  // for now skip validation for groups to avoid an exception. See #22933
+  try {
+    std::tie(inWS, indices) =
+        getWorkspaceAndIndices<API::MatrixWorkspace>(Prop::INPUT_WS);
+  } catch (...) {
```
Comment (samueljackson92) [inline Framework/Algorithms/src/ReflectometrySumInQ.cpp]: If there a more specific exception we can catch?

**Review by samueljackson92 — CHANGES_REQUESTED** [review]
Fix works. One minor thing:

File: Framework/Algorithms/src/ReflectometrySumInQ.cpp [hunk 2]
```diff
@@ -277,7 +277,7 @@ std::map<std::string, std::string> ReflectometrySumInQ::validateInputs() {
   try {
     std::tie(inWS, indices) =
         getWorkspaceAndIndices<API::MatrixWorkspace>(Prop::INPUT_WS);
-  } catch (...) {
+  } catch (std::runtime_error &) {
```
Comment (samueljackson92) [inline Framework/Algorithms/src/ReflectometrySumInQ.cpp]: 👍

**Review by samueljackson92 — APPROVED** [review]

**Comment by samueljackson92** [comment]
LGTM. Check the build passes and this is good to go.

**Pull request merged**